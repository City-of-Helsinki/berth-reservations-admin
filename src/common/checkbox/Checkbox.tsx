import React from 'react';
import classNames from 'classnames';

import { ReactComponent as Check } from './assets/check.svg';
import styles from './checkbox.module.scss';

export type CheckboxProps = Omit<
  React.InputHTMLAttributes<HTMLInputElement>,
  'size'
> & {
  label?: string;
  size?: 'small' | 'medium' | 'large';
};

const Checkbox: React.SFC<CheckboxProps> = ({
  checked,
  defaultChecked,
  disabled,
  onChange,
  label,
  size = 'medium',
  readOnly,
  id,
}) => {
  return (
    <label>
      <span
        className={classNames(styles.checkbox, styles[size], {
          [styles.checked]: checked,
          [styles.disabled]: disabled,
          [styles.readOnly]: readOnly,
        })}
      >
        {checked && <Check className={styles.check} />}
        <input
          id={id}
          checked={checked}
          defaultChecked={defaultChecked}
          onChange={onChange}
          className={styles.input}
          type="checkbox"
          readOnly={readOnly}
        />
      </span>
      {label && (
        <span
          className={classNames(styles.label, {
            [styles.disabled]: disabled,
            [styles.readOnly]: readOnly,
          })}
        >
          {label}
        </span>
      )}
    </label>
  );
};

export default Checkbox;
