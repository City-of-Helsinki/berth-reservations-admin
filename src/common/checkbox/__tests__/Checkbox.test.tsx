import React from 'react';
import { shallow } from 'enzyme';
import Checkbox from '../Checkbox';

const mockProps = {
  id: 'test',
};

describe('Checkbox', () => {
  const getWrapper = (props = {}) => shallow(<Checkbox {...mockProps} {...props} />);

  it('renders normally', () => {
    const wrapper = getWrapper();

    expect(wrapper.render()).toMatchSnapshot();
  });
});
