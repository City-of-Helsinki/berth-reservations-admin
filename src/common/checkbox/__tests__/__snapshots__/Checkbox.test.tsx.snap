// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox renders normally 1`] = `
<div
  class="Checkbox-module_checkbox__3L0GR checkbox_hds-checkbox__9HMCz checkbox"
>
  <input
    class="Checkbox-module_input__3VZvy checkbox_hds-checkbox__input__1w0pu"
    id="test"
    type="checkbox"
  />
  <label
    class="Checkbox-module_label__L5AN1 checkbox_hds-checkbox__label__3HoD3"
    for="test"
  />
</div>
`;
